unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BlockerController/Tests/BlockerHandlerCan.cs

```diff
@@ -0,0 +1,119 @@
+using DCL.Controllers;
+using DCL.Helpers;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.TestTools;
+using DCL;
+using System.Linq;
+using NSubstitute;
+using NUnit.Framework;
+
+public class BlockerHandlerCan
+{
+    private IBlockerInstanceHandler blockerInstanceHandler;
+    private IBlockerAnimationHandler blockerAnimationHandler;
+    private ISceneHandler sceneHandler;
+    private GameObject blockersParent;
+
+    [SetUp]
+    protected void SetUp()
+    {
+        RenderProfileManifest.i.Initialize();
+
+        sceneHandler = Substitute.For<ISceneHandler>();
+
+        var allLoadedParcelCoords = new HashSet<Vector2Int>();
+        allLoadedParcelCoords.Add(new Vector2Int(0, 0));
+        allLoadedParcelCoords.Add(new Vector2Int(-1, 0));
+        allLoadedParcelCoords.Add(new Vector2Int(-1, 1));
+
+        sceneHandler.GetAllLoadedScenesCoords().Returns(allLoadedParcelCoords);
+
+        blockerAnimationHandler = Substitute.For<IBlockerAnimationHandler>();
+
+        //NOTE(Brian): Call OnFinish() when blockerAnimationHandler.FadeOut is called. 
+        blockerAnimationHandler.FadeOut(Arg.Any<GameObject>(), Arg.Invoke());
+
+        blockerInstanceHandler = new BlockerInstanceHandler(new DCLCharacterPosition(), blockerAnimationHandler);
+
+        blockersParent = new GameObject();
+        blockerInstanceHandler.SetParent(blockersParent.transform);
+    }
+
+    [TearDown]
+    protected void TearDown()
+    {
+        Object.Destroy(blockersParent);
+    }
+
+    [Test]
+    public void ShowBlocker()
+    {
+        blockerInstanceHandler.ShowBlocker(new Vector2Int(0, 0));
+        blockerAnimationHandler.ReceivedWithAnyArgs(1).FadeIn(default);
+        Assert.AreEqual(1, blockerInstanceHandler.GetBlockers().Count);
+    }
+
+    [Test]
+    public void HideBlocker()
+    {
+        blockerInstanceHandler.ShowBlocker(new Vector2Int(0, 0), true);
+        blockerInstanceHandler.HideBlocker(new Vector2Int(0, 0), false);
+        blockerAnimationHandler.ReceivedWithAnyArgs(1).FadeOut(default, default);
+        Assert.AreEqual(0, blockerInstanceHandler.GetBlockers().Count);
+    }
+
+    [Test]
+    public void ShowBlockerInstantly()
+    {
+        blockerInstanceHandler.ShowBlocker(new Vector2Int(0, 0), true);
+        blockerAnimationHandler.DidNotReceiveWithAnyArgs().FadeIn(default);
+        Assert.AreEqual(1, blockerInstanceHandler.GetBlockers().Count);
+    }
+
+    [Test]
+    public void HideBlockerInstantly()
+    {
+        blockerInstanceHandler.ShowBlocker(new Vector2Int(0, 0), true);
+        blockerInstanceHandler.HideBlocker(new Vector2Int(0, 0), true);
+        blockerAnimationHandler.DidNotReceiveWithAnyArgs().FadeOut(default, default);
+        Assert.AreEqual(0, blockerInstanceHandler.GetBlockers().Count);
+    }
+
+    [Test]
+    public void SetParent()
+    {
+        // Prepare
```

Comment:
I can relate to this so much hahaha, the convention is `Arrange` instead of `Prepare` (ArrangeActAssert). I always forget what the first A stands for.